File: packages/engine/src/cpus/6502/instructions/dec.test.js
import { test, expect } from 'vitest'
import { BufferMapper } from '../../../buffer_mapper'
import { CPU } from '..'

test('dec', () => {
  const mmu = new BufferMapper()
  const cpu = new CPU(mmu)
  mmu.data = [0xc6, 0x2, 0x0] // DEC, $02
  const g = cpu.run()
  for (let i = 0; i < 5; i++) g.next()
  expect(mmu.data[0x2]).toBe(255)
})

File: packages/engine/src/cpus/6502/instructions/dec.test.zig
const testing = @import("../testing/setup.zig");

test "DEC" {
    var cpu = testing.createCPU();
    cpu.write(0, 0xc6);
    cpu.write(1, 2);
    cpu.write(2, 0);
    for (0..5) |_| {
        cpu.process();
    }
    try testing.expectEqual(255, cpu.read(2));
}
